_posts/jasonwang/2016-04-29-jasonwang-finalproject.md

```diff
@@ -0,0 +1,51 @@
+
+I decided to tackle this project by largely first finishing the individual data analysis parts, followed by the UI and consolidation process. To start, I focused on reading and processing the files. At this point, we had not discussed creating the project in Cloud 9, so I began in PyCharm to keep a degree of consistency between both IDE’s. However, PyCharm has an interesting format of in cases where tabs/spaces divide columns in text files, the perceived space increases (doubles?) for easier reading. Unfortunately, I did not know this prior, and thought that the transfer between original text file and PyCharm broke the formatting; leading to some wasted time. However, moving to Cloud 9 removed this problem.
+
+I then began data analysis. While I did have an idea on what I wanted to do, it was not until the beginning of data analysis where I was able to pinpoint on what results I wanted in the end. While I initially said that I wanted to compare certain values together, I thought about the meaning of such comparisons, and realized that overall, such outputs would not teach anyone much about data trends. However, general distributions, such as the average rating over time, or the average comment number over time, gives a picture of the Youtube landscape at the time. 
+
+I still took a modular approach, although a slightly different modular approach. I first set up a test sequence- where data flows freely between demarcated sections and produces the results. Once that was completed, I split the sections into separate functions (but still without data requesting). I really did think that the UI would be pretty easy to implement- but between the differences of the while loop and minimizing the data that flows between functions, the functions still had to be restructured (and new ones made). Perhaps in this case, I should have also made a rough UI version so that the functions that I had made were also designed to work with the While loops.
+
+Between this project and the last, I personally believe that my debugging skill had increased dramatically. While in the last project I had to keep playing around with values (and sometimes breaking it more) to try and isolate why certain functions weren’t working as intended, zeroing in on problems in this project was much easier. This may be helped by the fact that you can track the data as it flows through, but I also think that my deeper understanding of Python since the last project allowed me to stop and think, “Why?”. And from “why”, we can narrow down the possible areas of interest.
```

Comment:
+1 for introspection.  I think you've definitely got more skills. I also think those skills have come from your reflections on what you did before and that if you keep it up there's no limit to how far you can go.